src/ServiceControl.Plugin.Nsb6.Heartbeat/EnrichPreV44MessagesWithHostDetailsMutator.cs

```diff
@@ -0,0 +1,46 @@
+namespace ServiceControl.Plugin.Nsb6.Heartbeat
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Hosting;
+    using NServiceBus.MessageMutator;
+    using NServiceBus.Unicast;
+    using NServiceBus.Settings;
+
+    class EnrichPreV44MessagesWithHostDetailsMutator : IMutateIncomingTransportMessages
+    {
+        public EnrichPreV44MessagesWithHostDetailsMutator(ReadOnlySettings settings)
+        {
+            HostId = settings.Get<Guid>("NServiceBus.HostInformation.HostId");
+            Host = settings.Get<string>("NServiceBus.HostInformation.DisplayName");
+        }
+
+        public Task MutateIncoming(MutateIncomingTransportMessageContext transportMessage)
+        {
+            if (!transportMessage.Headers.ContainsKey("$.diagnostics.hostid"))
+            {
+                transportMessage.Headers["$.diagnostics.hostid"] = HostId.ToString();
+                transportMessage.Headers["$.diagnostics.hostdisplayname"] = Host;
+            }
+
+            return Task.FromResult(0);
+        }
+
+        Guid HostId;
+        string Host;
+    }
+
+    class EnrichPreV44MessagesWithHostDetailsMutatorInitializer : INeedInitialization
```

Comment:
Not sure about this. Couldn't constructor inject in the INeedInitialization instance, but needed the context to get the ReadOnlySettings, so split it out.

Is there a better way?